admin/app/controllers/SwitchboardPlistaController.scala

```diff
@@ -0,0 +1,94 @@
+package controllers.admin
+
+import com.gu.googleauth.UserIdentity
+import common._
+import common.AdminMetrics.{ SwitchesUpdateCounter, SwitchesUpdateErrorCounter }
+import conf.switches.Switches
+import controllers.AuthLogging
+import conf.Configuration
+import play.api.mvc._
+import scala.concurrent.Future
+import services.Notification
+import tools.Store
+import model.NoCache
+
+object SwitchboardPlistaController extends Controller with AuthLogging with Logging with ExecutionContexts {
+
+  val SwitchPattern = """([a-z\d-]+)=(on|off)""".r
+
+  def renderSwitchboard() = AuthActions.AuthActionTest.async { implicit request =>
+    log("loaded plista Switchboard", request)
+    Future { Store.getSwitchesWithLastModified } map { switchesWithLastModified =>
+      val configuration = switchesWithLastModified.map(_._1)
+      val nextStateLookup = Properties(configuration getOrElse "")
+
+      Switches.all foreach { switch =>
+        nextStateLookup.get(switch.name) foreach {
+          case "on" => switch.switchOn()
+          case _ => switch.switchOff()
+        }
+      }
+
+      val lastModified = switchesWithLastModified.map(_._2).map(_.getMillis).getOrElse(System.currentTimeMillis)
+      NoCache(Ok(views.html.switchboardPlista(Configuration.environment.stage, lastModified)))
+    }
+  }
+
+  def save() = AuthActions.AuthActionTest.async { implicit request =>
+    val form = request.body.asFormUrlEncoded
+
+    val localLastModified = form.get("lastModified").head.toLong
+    val remoteSwitches = Store.getSwitchesWithLastModified
+
+    // if any of the switches have a later modified date than locally, redirect
+    if (remoteSwitches.map(_._2).exists(_.getMillis > localLastModified)) {
+      Future {
+        NoCache(Redirect("/dev/switchboard-plista").flashing("error" -> "A more recent change to the switch has been found, please refresh and try again."))
+      }
+    } else {
+      log("saving plista switchboard", request)
+      val requester = UserIdentity.fromRequest(request).get.fullName
+      // for switches not present on this page, we need to persist their current values
```

Comment:
You only need the change in the plista switch and the remote switches that you've fetched.  You could make a map out of the remote switches and then add the plista switch to it, which would update the state of the plista switch.  Then save the new map in whatever form the method expects.